src/matchengine/controller.rs

```diff
@@ -552,6 +553,64 @@ impl Controller {
         Ok(())
     }
 
+    pub fn transfer(&mut self, real: bool, req: TransferRequest) -> Result<TransferResponse, Status> {
+        if !self.check_service_available() {
+            return Err(Status::unavailable(""));
+        }
+
+        let asset_id = req.asset;
+        if !self.balance_manager.asset_manager.asset_exist(&asset_id) {
+            return Err(Status::invalid_argument("invalid asset"));
+        }
+
+        let from_user_id = req.from;
+        let to_user_id = req.to;
+
+        let balance_manager = &self.balance_manager;
+        let balance_from = balance_manager.get(from_user_id, BalanceType::AVAILABLE, &asset_id);
+
+        let zero = Decimal::from(0);
+        let delta = Decimal::from_str(&req.delta).unwrap_or(zero);
+
+        if delta <= zero || delta > balance_from || !balance_manager.exists(to_user_id, BalanceType::AVAILABLE, &asset_id) {
```

Comment:
`!balance_manager.exists(to_user_id, BalanceType::AVAILABLE, &asset_id)`

sorry my bad.  It's possible to check user_to at the moment. I think it's fine to remove `|| !balance_manager.exists(to_user_id, BalanceType::AVAILABLE, &asset_id)` for now.